### ISSUE TITLE
NoClassDefFoundError: org/jetbrains/kotlin/gradle/dsl/KotlinProjectExtension

### ISSUE THREAD

**Issue by tprochazka** [Kotlin/dokka#1280]
After upgrade to docca 1.4.0-rc I'm getting an exception about missing `KotlinProjectExtension` class. I'm expecting that it is because a new version require apply `kotlin-android` plugin before dokka.  But I'm doing Dokka setup in own Gradle plugin which is common for Android project with and without Kotlin support. So far I was counting Dokka as universal Java/Kotlin doc generator for javadoc which automatically detect if Kotlin is present.

I'm not sure, but I think that it is cause by this code inside of dokkaSourceSets 
```
            // These tasks will be used to determine source directories and classpath
            kotlinTasks {
                defaultKotlinTasks() + listOf(
                    ":some:otherCompileKotlin",
                    project("another").tasks.getByName("compileKotlin")
                )
            }
```
It is default behavior if I have not custom setup right?
Would be possible to postpone this setup to after evaluate phase and detect if Kotlin is present please?

Here is full stacktrace
```
Caused by: java.lang.NoClassDefFoundError: org/jetbrains/kotlin/gradle/dsl/KotlinProjectExtension
        at org.jetbrains.dokka.gradle.ConfigurationExtractor.extractConfiguration(ConfigurationExtractor.kt:24)
        at org.jetbrains.dokka.gradle.DokkaTask.configureDokkaSourceSet(DokkaTask.kt:207)
        at org.jetbrains.dokka.gradle.DokkaTask.getConfigurationOrNull$gradle_plugin(DokkaTask.kt:151)
        at org.jetbrains.dokka.gradle.DokkaTask.getConfigurationOrThrow$gradle_plugin(DokkaTask.kt:171)
        at org.jetbrains.dokka.gradle.DokkaTask.generate(DokkaTask.kt:127)
        at org.jetbrains.dokka.gradle.AbstractDokkaTask.run(AbstractDokkaTask.kt:34)

```

**Comment by sellmair** [comment]
This mechanism will change quite some bit after #1194 gets merged. Can you re-check your issue after the mentioned pull request?

**Comment by tprochazka** [comment]
Thanks for reply! I wil check it.

**Comment by sellmair** [comment]
I think this issue can be closed for now, since the underlying code changed significantly and went under a lot of testing now. Feel free to re-open the issue if you still have questions/issues ☺️

**Comment by tprochazka** [comment]
I tried it again with final version, it not crash, but it always ends with empty folders and this log

```
> Task :sample-library:dokkaHtml
Dokka 1.4.* is an alpha project
Initializing plugins
Creating documentation models
Exiting Generation: Nothing to document

```
When I do it in the standard way by
```
plugins {
    id("org.jetbrains.dokka") version "1.4.0"
}

```
it works, even on Android.

But I have own Kotlin based Gradle plugin, which setup everything for all my projects (publishing, signing, documentation, etc.)

So I simply put this as the dependency of my plugin
```
implementation("org.jetbrains.dokka:dokka-gradle-plugin:1.4.0") 
implementation("org.jetbrains.dokka:dokka-core:1.4.0") // it was crashing without this
```
And in my plugin apply() method I call this
```
pluginManager.apply("org.jetbrains.dokka")
```

I'm not sure how plugins are handled in this way. When I use `plugins ` it works automatically. Somehow.

So I tried do also this

            pluginManager.apply("org.jetbrains.dokka")

            tasks.withType(DokkaTask::class.java).configureEach {
                dependencies {
                    plugins("org.jetbrains.dokka:dokka-base:1.4.0")
                    plugins("org.jetbrains.dokka:kotlin-as-java-plugin:1.4.0")
                }

But it doesn't help :-(

**Comment by tprochazka** [comment]
Key difference is the way how it is added to the classpath from some unknown reason.

When I move it from my plugin directly to the project build script as

```
buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath("org.jetbrains.dokka:dokka-gradle-plugin:1.4.0")
    }
}

```
everything works correctly.


But when I put it to the `buildSrc/build.gradle.kts`  (to simulate plugin behavior)

```
dependencies {
    implementation("org.jetbrains.dokka:dokka-gradle-plugin:1.4.0")  
}
```

It still works, dokka task is running, but it ends with `Exiting Generation: Nothing to document`

**Comment by tprochazka** [comment]
I rather created a new issue for it https://github.com/Kotlin/dokka/issues/1463
Including sample project demonstrating issue.
